From 5beae43938dba37e2e11e2ae150e501c1e5d3c34 Mon Sep 17 00:00:00 2001
From: Sathishkumar Muruganandam <quic_murugana@quicinc.com>
Date: Wed, 13 Dec 2023 18:20:16 +0530
Subject: [PATCH 1/2] wifi: mac80211: prevent KASAN warning in _find_next_bit

for_each_set_bit() invoking _find_next_bit expects only unsigned long arguments
which causes KASAN warning for u16 usage of valid_links variable.

Fix this by assigning valid_links as unsigned long argument to
for_each_set_bit().

Signed-off-by: Sathishkumar Muruganandam <quic_murugana@quicinc.com>
---
 net/mac80211/rx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 26d6896964f4..463e4ccd53e6 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -5530,7 +5530,7 @@ static void __ieee80211_rx_handle_packet(struct ieee80211_hw *hw,
 	list_for_each_entry_rcu(sdata, &local->interfaces, list) {
 		unsigned int link_id;
 		struct ieee80211_bss_conf *conf;
-		u16 valid_links = sdata->vif.valid_links;
+		unsigned long valid_links = sdata->vif.valid_links;
 		bool flag = false;
 
 		if (!ieee80211_sdata_running(sdata))
@@ -5604,7 +5604,7 @@ static void __ieee80211_rx_handle_packet(struct ieee80211_hw *hw,
 		if (prev) {
 			unsigned int link_id;
 			struct ieee80211_bss_conf *conf;
-			u16 valid_links = prev->vif.valid_links;
+			unsigned long valid_links = prev->vif.valid_links;
 			bool flag = false;
 
 			if (valid_links) {
-- 
2.25.1

